### PR TITLE
Add 2023 Virginia Tax Rules

### LINKED ISSUE #4199: Add 2023 Virginia Tax Rules
2023 Income Tax Form Instructions: https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2023-760-instructions.pdf

### PR DESCRIPTION
**Pull request by Mshuning**
Fixes #4199

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
Could you summarize the changes in a comment here?

**Comment by Mshuning** [comment]
Only itemized deduction amount has changed in 2023.

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: Could you take a look at this reference:
https://www.tax.virginia.gov/sites/default/files/inline-files/2023-legislative-summary.pdf

In the income tax section it talks about a new rebate, not sure if it is in the 2023 tax form

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (Mshuning) [reply changelog_entry.yaml:1]: This doesn't seem to appear in https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2023-760-instructions.pdf

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: I think it only comes into effect in the middle of the year? I wonder if it wont show up on the tax form because of that, could you take a look at the legal code and see if it is there?

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_basic_pay/threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_basic_pay/threshold.yaml:1]: Could we also model this for year after 2024:
image

The age requirement seems to disappear

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/age_threshold.yaml:1]: Could we also model this for year after 2024:
image

The age requirement seems to disappear

File: changelog_entry.yaml
Comment (Mshuning) [reply changelog_entry.yaml:1]: Related laws are listed as follows: https://budget.lis.virginia.gov/item/2023/2/HB6001/Introduced/3/3-5.28/
image
But couldn't find any law mentioned this rebate from https://law.lis.virginia.gov/vacode/title58.1/chapter3/

Comment (Mshuning) [reply changelog_entry.yaml:1]: maybe helpful: https://www.tax.virginia.gov/rebate

File: policyengine_us/parameters/gov/states/va/tax/income/deductions/itemized/applicable_amount.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ JOINT:
   2019-01-01: 326_050 # 2020 no changes
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/deductions/itemized/applicable_amount.yaml:9]: Do these amounts get updated with inflation? 
If so we will need to add some metadata to represent that,
See DC PR:
https://github.com/PolicyEngine/policyengine-us/pull/4306

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/va/tax/income/deductions/itemized/applicable_amount.yaml:9]: same for other files which seem to be updated yearly, we will need to check the legal code for that

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 2]
```diff
@@ -21,11 +21,15 @@ def formula(tax_unit, period, parameters):
         subtractable_military_benefit = min_(
             person("military_retirement_pay", period), p.amount
         )
-
-        age = person("age", period)
-        head_or_spouse = person("is_tax_unit_head_or_spouse", period)
-        age_eligible = age >= p.age_threshold
-        head_or_spouse_eligible = head_or_spouse & age_eligible
+        if p.availability:
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:24]:
```suggestion
        head_or_spouse = person("is_tax_unit_head_or_spouse", period)
        if p.availability:
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 3]
```diff
@@ -21,11 +21,15 @@ def formula(tax_unit, period, parameters):
...
+            age = person("age", period)
+            head_or_spouse = person("is_tax_unit_head_or_spouse", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:26]:
```suggestion
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 4]
```diff
@@ -21,11 +21,15 @@ def formula(tax_unit, period, parameters):
...
+            age_eligible = age >= p.age_threshold
+            head_or_spouse_eligible = head_or_spouse & age_eligible
+        else:
+            head_or_spouse_eligible = person(
+                "is_tax_unit_head_or_spouse", period
+            )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:32]:
```suggestion
            head_or_spouse_eligible = head_or_spouse

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/availability.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+description: Virginia allows for military benefit subtraction to be claimed regardless of age, effective after this year.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/availability.yaml:1]:
```suggestion
description: Virginia allows for the military benefit subtraction to be claimed regardless of age, effective after this year.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/va/tax/income/rebate/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+description: Virginia issues a refund out of the Individual Income Tax Rebate Fund in this amount, based on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/rebate/amount.yaml:1]:
```suggestion
description: Virginia provides an individual income tax rebate of this amount, based on filling status.
```

File: policyengine_us/parameters/gov/states/va/tax/income/rebate/availability.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+description: Virginia issues a refund out of the Individual Income Tax Rebate Fund in this year.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/rebate/availability.yaml:1]:
```suggestion
description: Virginia allows for an individual income tax rebate if this is true.
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/availability.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+description: Virginia allows for the military benefit subtraction to be claimed regardless of age, effective after this year.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/availability.yaml:1]:
```suggestion
description: Virginia allows for the military benefit subtraction to be claimed regardless of age if this is true.
```

File: policyengine_us/variables/gov/states/va/tax/income/va_rebate.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    unit = USD
+    definition_period = YEAR
+    reference = "https://budget.lis.virginia.gov/item/2023/2/HB6001/Introduced/3/3-5.28/"
+    defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.va.tax.income.rebate
+        if p.availability:
+            filing_status = tax_unit("filing_status", period)
+            return p.amount[filing_status]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:17]:
```suggestion
        p = parameters(period).gov.states.va.tax.income.rebate
        if p.availability:
            filing_status = tax_unit("filing_status", period)
            return p.amount[filing_status]
         return 0 
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/va/tax/income/credits/non_refundable.yaml [hunk 10]
```diff
@@ -2,11 +2,14 @@ description: Virginia non-refundable tax credits.
 values:
   2021-01-01:
     - va_non_refundable_eitc
+    - va_rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/credits/non_refundable.yaml:5]:
```suggestion
  2021-01-01:
    - va_non_refundable_eitc
  2023-01-01:
    - va_non_refundable_eitc
    - va_rebate
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/va/tax/income/deductions/standard.yaml [hunk 11]
```diff
@@ -8,10 +8,12 @@ metadata:
   reference:
   - title: Code of Virginia § 58.1-322.03.(1)(b)
     href: https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/
+  - title: Virginia 2023 Form 760 Resident Individual Income Tax Instructions
+    href: https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2023-760-instructions.pdf#page=18
   - title: Virginia 2022 Form 760 Resident Individual Income Tax Booklet
     href: https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2022-760-instructions.pdf#page=18
   - title: Virginia 2021 Form 760 Resident Individual Income Tax Booklet
-    href: https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2021-760-instructions.pdf#page=19
+    href: https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2021-760-instructions.pdf#page=18
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/deductions/standard.yaml:16]: could you add the 2024 - 2026 standard deduction changes while in here?
[Link](https://www.tax.virginia.gov/sites/default/files/inline-files/2023-legislative-summary.pdf#page=10)

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 12]
```diff
@@ -20,6 +20,8 @@ metadata:
   period: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:20]: Could you take a look at this line:
`    - military_service_income # Certification Number 28
`
Military service income as used as national guard pay and I think we subtract all of it but it should be capped - even before 2023 
image

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_rebate.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+        return 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]:
```suggestion
    def formula(tax_unit, period, parameters):
        p = parameters(period).gov.states.va.tax.income.rebate
        filing_status = tax_unit("filing_status", period)
        return p.amount[filing_status]
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]: I made a mental mistake, if we add it to the 2023 non refundable credits then we by design dont need the if condition

File: policyengine_us/parameters/gov/states/va/tax/income/rebate/availability.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/rebate/availability.yaml:1]: delete file - dont need it

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml [hunk 14]
```diff
@@ -0,0 +1,20 @@
+description: Virginia provides a national guard pay subtraction up to this amount.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml:1]:
```suggestion
description: Virginia caps the national guard pay subtraction to this amount.
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+values:
+  2021-01-01: 3_000
+  2023-01-01: 5_500
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Virginia national guard pay subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml:9]:
```suggestion
  label: Virginia national guard pay subtraction cap
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/national_guard_income.yaml:1]: make a `national_guard_pay` folder and rename this `cap.yaml`

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_national_guard_subtraction.py [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
+    )
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        p = parameters(period).gov.states.va.tax.income.subtractions
+
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_national_guard_subtraction.py:18]:
```suggestion
      national_guard_pay = add(tax_unit, period, ["military_service_income"])
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_national_guard_subtraction.py [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    label = "Virginia national guard pay subtraction"
+    defined_for = StateCode.VA
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
+    )
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_national_guard_subtraction.py:16]:
```suggestion
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_rebate.py (hunk 13 above)
Comment (Mshuning) [reply policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]: but then the `va_rebate.yaml` will report error with 2022 test cases

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]: Yes but we never pass it through the 2022 simulation if we only include it in the 2023 period non_refundable file

Comment (Mshuning) [reply policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]: so then should I delete 2022 test case in `va_rebate.yaml`?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_rebate.py:18]: Yes please

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]
